zerver/tests/test_message_fetch.py

```diff
@@ -1310,6 +1310,17 @@ def verify(
 
 
 class GetOldMessagesTest(ZulipTestCase):
+    def setUp(self) -> None:
+        super().setUp()
+        self.realm = get_realm("zulip")
+        self.realm.enable_spectator_access = True
+        self.realm.save()
+
+    def tearDown(self) -> None:
+        super().tearDown()
+        self.realm.enable_spectator_access = False
+        self.realm.save()
```

Comment:
A few points:
* We should use `do_set_realm_propert` rather than raw updates to the realm object, since that more faithfully simulates actual runtime behavior.  I've been meaning to do a sweet of this in the tests and then add a linter rule for it.
* The tearDown here should be unnecessary because of the transaction setup noted in another comment.

Reply:
Ah! Thanks.